CMakeLists.txt

```diff
@@ -32,6 +32,8 @@ if (MSVC)
   # Use add_compile_options() to set /MT since Visual Studio
   # Generator does not notice /MT in CMAKE_C_FLAGS.
   add_compile_options(/MT)
+elseif(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -fno-common")
```

Comment:
Without this, linking failed with below error.
```
[ 85%] Linking C executable ../bin/fluent-bit
Undefined symbols for architecture x86_64:
  "_flb_thread_key", referenced from:
      _flb_main in fluent-bit.c.o
      _flb_engine_start in libfluent-bit.a(flb_engine.c.o)
      _flb_input_collector_fd in libfluent-bit.a(flb_input.c.o)
      _flb_engine_dispatch_retry in libfluent-bit.a(flb_engine_dispatch.c.o)
      _tasks_start in libfluent-bit.a(flb_engine_dispatch.c.o)
      _flb_output_return_do in libfluent-bit.a(flb_plugin_proxy.c.o)
      _create_conn in libfluent-bit.a(flb_upstream.c.o)
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Although symbol `_flb_thread_key` is indeed included in libfluent-bit.a
```
❯ nm ./library/libfluent-bit.a | rg flb_thread_key
                 U _flb_thread_key
                 U _flb_thread_key
                 U _flb_thread_key
                 U _flb_thread_key
                 U _flb_thread_key
                 U _flb_thread_key
0000000000000008 C _flb_thread_key
                 U _flb_thread_key
```
As suggested by [this thread](https://github.com/libigl/libigl/issues/751#issuecomment-383324059), using flag `-fno-common` will enforce the uninitialized variable not be put in common section.

```
❯ nm ./library/libfluent-bit.a| rg flb_thread_key
                 U _flb_thread_key
                 U _flb_thread_key
                 U _flb_thread_key
                 U _flb_thread_key
                 U _flb_thread_key
                 U _flb_thread_key
0000000000000080 S _flb_thread_key
                 U _flb_thread_key
```

Reply:
I think problem with in  /include/fluent-bit/flb_thread.h:37

Line: 
`extern FLB_EXPORT pthread_key_t flb_thread_key`

Should be:
`FLB_EXPORT pthread_key_t flb_thread_key;`

Reply:
Thanks for the pointer, @bluebike! That's indeed what caused the issue, introduced by https://github.com/fluent/fluent-bit/commit/328982390b04e81f459f742362dbff56bb6dcb54

PR updated with the fix you suggested.

Reply:
@edsiper I restored my initial attempted solution that append flag `-fno-common` on Darwin.

clang has made[ `-fno-common` default](https://github.com/llvm/llvm-project/commit/0a9fc9233e172601e26381810d093e02ef410f65) to be aligned with GCC 10. It'll be included in the next release.